pydantic_extra_types/coordinate.py

```diff
@@ -0,0 +1,202 @@
+from typing import Any, Callable, ClassVar, Tuple, Type, Union
+
+from pydantic import GetCoreSchemaHandler, GetJsonSchemaHandler
+from pydantic._internal import _repr
+from pydantic.json_schema import JsonSchemaValue
+from pydantic_core import CoreSchema, PydanticCustomError, core_schema
+
+LatitudeType = Union[str, int, float]
+
+
+class Latitude(float):
+    ge: ClassVar[float] = -90.00
+    le: ClassVar[float] = 90.00
+
+    def __init__(self, value: LatitudeType):
+        self.validate_lat(value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls._validate, core_schema.float_schema(ge=cls.ge, le=cls.le)
+        )
+
+    @classmethod
+    def _validate(cls, __input_value: Any, _: core_schema.ValidationInfo) -> 'Latitude':
+        return cls(__input_value)
+
+    @classmethod
+    def validate_lat(cls, value: LatitudeType) -> None:
+        """
+        Validate the latitude value.
+
+        Raises:
+            PydanticCustomError: If the latitude value is not within the valid range.
+        """
+        try:
+            if float(value) > cls.le or float(value) < cls.ge:
+                raise PydanticCustomError('latitude_error', 'Latitude must be between -90 and 90')
+        except ValueError:
+            raise PydanticCustomError(
+                'latitude_error',
+                'value is not a valid Latitude: value must be a str, int or float',
+            )
+
+
+LongitudeType = Union[str, int, float]
```

Comment:
Could we use the same type for Latitude and Longitude? I see no reason to not to.

Reply:
I think there are better options for the name of this type.

- Usually Latitude is being named before Longitude.
- Also, we could stress here that this is a value and that it has nothing to do with being Latitude or Longitude.

How about `CoordinateValueType`?

Reply:
Done, thanks